tests/acceptance/test_mender-connect.py

```diff
@@ -193,14 +202,18 @@ def test_mender_connect_auth_changes(
         """Test that mender-connect can re-establish the connection on D-Bus signals"""
 
         try:
-            dbus_set_token_and_url(connection, "token1", "http://localhost:5000")
-
             # start the mender-connect service
-            startup_time = qemu_system_time(connection)
             connection.run(
                 "systemctl --job-mode=ignore-dependencies start mender-connect"
             )
 
+            dbus_set_token_and_url(connection, "token1", "http://localhost:5000")
+
+            # sleep for a while to ensure a connection happens
+            time.sleep(60)
+            dbus_emit_signal(connection)
+            signal_time = qemu_system_time(connection)
+
             # wait for first connect
             _ = wait_for_string_in_log(
```

Comment:
The sleep here looks suspicious to me. The `wait_for_string_in_log` function below will wait a given amount of time for the required message, so we shouldn't need to wait before.

You probably meant to wait for `mender-connect` to start-up before we can instruct the mock to emit the signal, so that will be short sleep between `systemctl ... start mender-connect` and `dbus_emit_signal`.

A much better approach that we draft some time ago on the whiteboard would be to have a flag in the mock that can save when `mender-connect` has called `FetchJwtToken` and then the test can wait for that to then emit the signal. This is a bit more complex but will be a better test (right now we are blindly assuming that `mender-connect` requested a token before we emit the signal).